src/common/FastRational.h

```diff
@@ -245,6 +247,16 @@ class FastRational
         }
     }
 
+    std::optional<std::pair<word, uword>> tryGetNumDen() const {
+        if (!wordPartValid()) return {};
+        return std::make_pair(num, den);
+    }
+
+    mpq_ptr getMpq() const {
```

Comment:
I am wondering if this is the right API?
Should we instead return a value, i.e., mpq_class?
So that the user does not do anything stupid like calling `mpq_clear` on the returned `mpq_ptr` and with that corrupting our storage.
It would also allow us to *not* use the problematic `force_ensure_mpq_valid` method.

Reply:
I agree, I somehow thought that `mpq_ptr` is sth. like `unique_ptr` but it is just a pointer without taking the ownership..
It seems to me that `mpq_class` should be a good choice, but I am not sure how to construct it. I tried to look up a constructor from `mpq_ptr` but I failed - although I found that it is used at `FastRational.cc:84`. But I do not know what exactly it does. Still it seems that it requires `mpqPartValid()`, so I suppose that `force_ensure_mpq_valid` is still necessary.